src/parsers.rs

```diff
@@ -510,6 +529,11 @@ mod test {
         // Long number
         assert_eq!(Ipv6Address::from_str("::000001"),
                    Err(()));
+        // IPv4-Mapped address
+        assert_eq!(Ipv6Address::from_str("::ffff:192.168.1.1"),
+                   Ok(Ipv6Address([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0xff, 0xff, 192, 168, 1, 1])));
+        assert_eq!(Ipv6Address::from_str("0:0:0:0:0:FFFF:192.168.1.1"),
+                   Ok(Ipv6Address([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0xff, 0xff, 192, 168, 1, 1])));
```

Comment:
Could you add some cases that fail? E.g. `::eeee:192.168.1.1`?

Reply:
I *think* `::1:fff:192.168.1.1` would also be accepted as a valid IPv4 mapped IPv6 address under the current parsing rules.

Reply:
done

Reply:
Actually my example was bad. `0:0:0:0:1:FFFF:192.168.1.1` is still parsed as a correctly formatted IPv4 mapped IPv6 address, but it shouldn't be. I think we need to check to ensure that the value of the head part contains only zero.

Reply:
Ah, yes. i updated the code to make sure that positions 0-4 of the head are 0. I also added some more test cases that fail.